File: laundry-reservation/typescript/tests/RecordingSmsNotifier.ts
import type { SmsNotifier } from '../src/SmsNotifier.js';

export interface SmsMessage {
  to: string;
  message: string;
}

export class RecordingSmsNotifier implements SmsNotifier {
  readonly sent: SmsMessage[] = [];

  send(to: string, message: string): void {
    this.sent.push({ to, message });
  }
}

File: laundry-reservation/python/tests/recording_sms_notifier.py
from __future__ import annotations

from dataclasses import dataclass


@dataclass
class SmsMessage:
    to: str
    message: str


class RecordingSmsNotifier:
    def __init__(self) -> None:
        self.sent: list[SmsMessage] = []

    def send(self, to: str, message: str) -> None:
        self.sent.append(SmsMessage(to, message))
